util/eventbus/eventbustest: TestExpectExactlyEventsFilter test flake

Flaky test seen at https://github.com/tailscale/tailscale/actions/runs/17848209515/job/50751740323?pr=17189

```
=== RUN   TestExpectExactlyEventsFilter
=== RUN   TestExpectExactlyEventsFilter/No_expectations
--- PASS: TestExpectExactlyEventsFilter/No_expectations (0.00s)
=== RUN   TestExpectExactlyEventsFilter/One_event
--- PASS: TestExpectExactlyEventsFilter/One_event (0.00s)
=== RUN   TestExpectExactlyEventsFilter/Two_events
--- PASS: TestExpectExactlyEventsFilter/Two_events (0.00s)
=== RUN   TestExpectExactlyEventsFilter/Two_expected_events_with_another_in_the_middle
--- PASS: TestExpectExactlyEventsFilter/Two_expected_events_with_another_in_the_middle (0.00s)
=== RUN   TestExpectExactlyEventsFilter/Missing_event
--- PASS: TestExpectExactlyEventsFilter/Missing_event (0.00s)
=== RUN   TestExpectExactlyEventsFilter/One_event_with_value
    eventbustest_test.go:407: ExpectEvents: error = expected event type eventbustest_test.EventFoo, saw eventbustest_test.EventBaz, at index 0, wantErr false
--- FAIL: TestExpectExactlyEventsFilter/One_event_with_value (0.00s)
=== RUN   TestExpectExactlyEventsFilter/Two_event_with_one_specific_value
--- PASS: TestExpectExactlyEventsFilter/Two_event_with_one_specific_value (0.00s)
=== RUN   TestExpectExactlyEventsFilter/One_event_with_wrong_value
--- PASS: TestExpectExactlyEventsFilter/One_event_with_wrong_value (0.00s)
=== RUN   TestExpectExactlyEventsFilter/Two_events_with_specific_values
--- PASS: TestExpectExactlyEventsFilter/Two_events_with_specific_values (0.00s)
--- FAIL: TestExpectExactlyEventsFilter (0.00s)
FAIL
FAIL	tailscale.com/util/eventbus/eventbustest	0.066s
```

/cc @creachadair @cmol @danderson

--- a/util/eventbus/eventbustest/eventbustest_test.go
+++ b/util/eventbus/eventbustest/eventbustest_test.go
@@ -108,10 +108,11 @@ func TestExpectFilter(t *testing.T) {
 		},
 	}
 
-	bus := eventbustest.NewBus(t)
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			bus := eventbustest.NewBus(t)
+			t.Cleanup(bus.Close)
+
 			if *doDebug {
 				eventbustest.LogAllEvents(t, bus)
 			}
@@ -241,10 +242,11 @@ func TestExpectEvents(t *testing.T) {
 		},
 	}
 
-	bus := eventbustest.NewBus(t)
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			bus := eventbustest.NewBus(t)
+			t.Cleanup(bus.Close)
+
 			tw := eventbustest.NewWatcher(t, bus)
 			// TODO(cmol): When synctest is out of experimental, use that instead:
 			// https://go.dev/blog/synctest
@@ -374,10 +376,11 @@ func TestExpectExactlyEventsFilter(t *testing.T) {
 		},
 	}
 
-	bus := eventbustest.NewBus(t)
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			bus := eventbustest.NewBus(t)
+			t.Cleanup(bus.Close)
+
 			tw := eventbustest.NewWatcher(t, bus)
 			// TODO(cmol): When synctest is out of experimental, use that instead:
 			// https://go.dev/blog/synctest

